File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int slow = 0;
        int fast = 0;
        int count = 0;

        while(fast < nums.size()){
            if(nums[fast] == 1) fast++;
            else {
                fast++;
                slow = fast;
            }
            count = max(fast-slow, count);
        }

        return count;
    }
};

File: 485-max-consecutive-ones/max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {0
    let fast = 0;
    let count = 0;
    let res = 0;

    while(fast < nums.length){
        if(nums[fast] === 1){
            count++;
            res = Math.max(res, count);
        }else{
            count = 0;
        }
        fast++;
    }

    return res;
};